packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1269,8 +1284,10 @@ class _TabBarState extends State<TabBar> {
       indicatorPadding: widget.indicatorPadding,
       tabKeys: _tabKeys,
       old: _indicatorPainter,
-      dividerColor: theme.useMaterial3 ? widget.dividerColor ?? tabBarTheme.dividerColor ?? _defaults.dividerColor : null,
       labelPaddings: _labelPaddings,
+      dividerColor: theme.useMaterial3 ? widget.dividerColor ?? tabBarTheme.dividerColor ?? _defaults.dividerColor : null,
+      dividerHeight: theme.useMaterial3 ? widget.dividerHeight ?? tabBarTheme.dividerHeight ?? _defaults.dividerHeight : null,
+      width: MediaQuery.sizeOf(context).width,
```

Comment:
@TahaTesser do you recall why this changed? I do not remember since this PR has had so much back and forth, I am looking though..

Reply:
You mean the `width`? This what divider painters uses to draw the divider beyond the tabs without changing the layout.

Reply:
Previously divider was only drawn with in the tabs, this caused this issue. https://github.com/flutter/flutter/issues/117722

Reply:
Right right. Thank you for the reference! I could not recall. We're reviewing all of the screenshots one last time and there was a weird one where the customer mocked up a phone outline in the test and the divider extended out beyond it. I think it is an unintended issue of the test, not this change. No worries. :)

Reply:
Ok so sad news, I am going to revert this. Unfortunately this is not uncommon when there are over a thousand screenshots. 😢 

I misunderstood what an image was capturing. The width here actually causes an issue when the TabBar is contained within a ConstrainedBox, the divider extends beyond it.

The customer test was not showing a mocked up phone, but rather a floating box with the content it was floating over omitted from the test case.

I've put together a repro here:

![Screenshot 2023-06-22 at 5 28 26 PM](https://github.com/flutter/flutter/assets/16964204/fb8afc61-2759-4bcc-82ae-4b3b5d9006c5)


```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {

    return MaterialApp(
      theme: ThemeData(useMaterial3: true),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {

  const MyHomePage({ super.key });

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: DefaultTabController(
        length: 2,
        child: Center(
          child: ConstrainedBox(
            constraints: const BoxConstraints(maxWidth: 360),
            child: ColoredBox(
              color: Colors.grey[200]!,
              child: const TabBar.secondary(
                tabAlignment: TabAlignment.start,
                isScrollable: true,
                tabs: [
                  Tab(text: 'Test 1'),
                  Tab(text: 'Test 2'),
                ],
              ),
            )
          ),
        ),
      ),
    );
  }
}
```